sys/include/random.h

```diff
@@ -34,14 +34,6 @@
 extern "C" {
 #endif
 
-#ifndef RANDOM_SEED_DEFAULT
-/**
- * @brief   Seed selected when all tries to collect seeds from a random source
- *          failed
- */
-#define RANDOM_SEED_DEFAULT (1)
-#endif
```

Comment:
I'm not sure if we should simply remove this block.
(i) It prevents users to set their own fallback seed. It is a relevant use case? Yes, for debugging purposes. However, in that case, we would want a static value over all other seed sources anyway, which was never the case. This requires (minimal) additional configuration effort **but*  it could justify to always randomize the static fallback... 
(ii) It removes documentation of that parameter.

What do you think @nmeum?

Reply:
Don't have any strong opinions on this since it's easy to implement and doesn't have any obvious disadvantages. I am fine with either option.

Reply:
Though one obvious disadvantage would be that `RANDOM_SEED_DEFAULT` would be set to a constant value without anyone noticing.

Reply:
I'm wondering if anyone has ever even used this parameter? The two options that I see here are as follows:
- Simply remove the parameter because it is "nonsense". It is not "nonsense" for debugging purposes, but a dedicated debug mode would be implemented differently anyway.
- Add a switch so that either `RANDOM_SEED_DEFAULT` or *randhex* can be used. 

Opinions @nmeum, @leandrolanzieri?

Reply:
> I'm wondering if anyone has ever even used this parameter?

I think its unlikely, especially since the parameter is only used if no other seed generation facility is available. I think the interesting question is: In which case does it make sense to supply a hardcoded seed for random number generation. Debugging is certainly a scenario where this might be useful but in that case `RANDOM_SEED_DEFAULT` must take precedence over the other facilities. For debugging purposes it might even be sufficient to simply modify  the `init.c` source. 

IMHO we could remove `RANDOM_SEED_DEFAULT` entirely for now, and maybe add a "dedicated debug mode" if it turns out to be needed or is explicitly requested by developers?

Reply:
Yes, that is what I meant with "dedicated debug mode would be implemented differently anyway" so I agree with you. Let's wait 1-2 days with merging so others still have time to state their opinions.

In the meantime, how about we add a compile time warning for the case that a proper entropy source is missing (like [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/puf_sram/Makefile#L1))?

Reply:
> In the meantime, how about we add a compile time warning for the case that a proper entropy source is missing (like here)?

Sounds good, will do that tomorrow! :blush:

Reply:
Added a warning to the makefile.